deepinv/tests/test_external_libraries.py

```diff
@@ -177,3 +178,39 @@ def test_tomography_with_astra_logic(
             assert physics.xray_transform.magnification_factor == pytest.approx(1.25)
         else:
             assert physics.xray_transform.magnification_factor == pytest.approx(1.0)
+
+
+class TestSPECT:
+    @pytest.mark.parametrize("use_att", [False, True])
+    @pytest.mark.parametrize("use_psf", [False, True])
+    def test_adjoint(self, use_att, use_psf):
+        from pytomography.metadata.SPECT import SPECTObjectMeta, SPECTProjMeta, SPECTPSFMeta
+
+        device = "cuda" if torch.cuda.is_available() else "cpu"
+
+        object_meta = SPECTObjectMeta(dr = (1, 1, 1), 
+                              shape = [32,32,32])
+        proj_meta = SPECTProjMeta(projection_shape= [32,32], 
+                          dr= (1, 1),
+                          angles= np.linspace(0, 360, 120, endpoint=False).tolist(), 
+                          radii=[380] * 120)
+        
+        # --- Optional transforms ---
+        attmap = torch.rand([32,32,32]) if use_att else None
+        psf_meta = SPECTPSFMeta(sigma_fit_params= np.random.rand(2)) if use_psf else None
+        physics = dinv.physics.SPECT(object_meta, proj_meta, psf_meta = psf_meta, attmap = attmap, device=device)
+
+        ## --- Test adjointness ---
```

Comment:
This is very cool thanks!

The code here is essentially following that of `physics.adjointness_test`. Maybe we could modify the code there and use that? We could modify it to have:

- a "normalize" parameter which divides by xTATy, which you do here;
- option to use `deepinv.utils.tensorlist.rand_like` rather than `randn_like`, which you do here, for data>0
- Then you could directly integrate this test into `test_physics`. Can we achieve 1e-6 adjointness error?